sdk/storage/Azure.Storage.Blobs.Cryptography/src/LengthLimitingStream.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage.Blobs.Specialized.Cryptography
+{
+    using System;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// A read-only stream. Wraps a <see cref="Stream"/> with an upper limit on how many bytes can be read.
+    /// </summary>
+    internal class LengthLimitingStream : Stream
```

Comment:
Nit - I don't love the name of this because the length limiting is completely optional.  It's more that this is a fixed readonly window of a Stream.

Reply:
Do you have an alternative? This name makes sense to me; it applies an optional limit to the length of content this stream can return.